lib/summarization/strategies/hot_topic_gists.rb

```diff
@@ -13,20 +13,21 @@ def targets_data
 
           op_post_number = 1
 
-          last_twenty_posts =
+          hot_topics_recent_cutoff = Time.zone.now - SiteSetting.hot_topics_recent_days.days
+
+          recent_hot_posts =
             Post
               .where(topic_id: target.id)
               .where("post_type = ?", Post.types[:regular])
               .where("NOT hidden")
-              .order("post_number DESC")
-              .limit(20)
+              .where("created_at >= ?", hot_topics_recent_cutoff)
               .pluck(:post_number)
 
```

Comment:
```suggestion

          # It may happen that a topic is hot without any recent posts
          # In that case, we'll just grab the last 20 posts
          # for an useful summary of the current state of the topic
          if recent_hot_posts.empty?
            recent_hot_posts =
              Post
                .where(topic_id: target.id)
                .where("post_type = ?", Post.types[:regular])
                .where("NOT hidden")
                .order("post_number DESC")
                .limit(20)
                .pluck(:post_number)
          end
```